tools/config/go-templates/api.mustache

```diff
@@ -216,7 +216,10 @@ func (a *{{{classname}}}Service) {{nickname}}Execute(r {{#structPrefix}}{{&class
 	{{^isCollectionFormatMulti}}
 		parameterAddToHeaderOrQuery(localVarQueryParams, "{{baseName}}", r.{{paramName}}, "{{collectionFormat}}")
 	{{/isCollectionFormatMulti}}
-	}
+	{{#defaultValue}}} else {
+		var defaultValue {{{dataType}}} = {{{.}}}
+		r.{{paramName}} = &defaultValue
+	{{/defaultValue}}}
```

Comment:
just for me to understand, why do we need to set the default? I was assuming the problem was because of not using pointers, and in which case it was defaulting to false. I would expect that while using a pointer and having the omit empty instruction the parameter won't get serialized then default to what the API defaults, is this not how it works for the generated SDK?

Reply:
I hope I understand question:

Query parameters aren't serialized - they are parsed into request url.
Required query params are always on method to create request. 
Optional ones are in struct and pointers which might be confusing.

Adding default support gives us ability to improve contract for optional parameters (omit empty does not apply).
The rest is like you have mentioned - if someone did not provided value we do check if default is present and assign it.

Reply:
>Query parameters aren't serialized 

Ok this may be the gap I may have, we were using a google lib that let's you define query params the same as a you do for json, and it applies the same rules as for json and nnotations, https://github.com/google/go-querystring seems like I need to give how openapi-gen does this a look to get familiar with it